File: cf_sdk/src/main/java/com/example/cf_sdk/logic/sdk_module/creator/MemberApiCreator.java
package com.example.cf_sdk.logic.sdk_module.creator;

import com.example.cf_sdk.logic.changebankapi.network.api.ApiCreator;
import com.example.cf_sdk.logic.changebankapi.network.api.MemberApi;
import com.example.cf_sdk.logic.changebanksdk.log.Logger;

import retrofit2.Retrofit;

/**
 *
 * Network {@link MemberApi} creator.
 */

public class MemberApiCreator implements ApiCreator<MemberApi> {
    @Override
    public MemberApi create(Logger logger, Retrofit retrofit) {
        return retrofit.create(MemberApi.class);
    }
}

File: cf_sdk/src/main/java/com/example/cf_sdk/changebankapi/network/api/MemberApiCreator.kt
package com.example.cf_sdk.changebankapi.network.api


import com.example.cf_sdk.changebankapi.log.Logger
import com.example.cf_sdk.changebankapi.network.api.MemberApi
import retrofit2.Retrofit

/**
 *
 * Network [MemberApi] creator.
 */
class MemberApiCreator : ApiCreator<MemberApi?> {
    override fun create(logger: Logger?, retrofit: Retrofit?): MemberApi? {
        return retrofit?.create(MemberApi::class.java)
    }
}